static-analysis/teammates-macker.xml

```diff
@@ -1,45 +1,399 @@
-        <access-rule>
-            <deny>
-                <from class="teammates.test.**" />
-                <to class="teammates.test.pageobjects.*" />
-                <allow>
-                    <!-- allow dependency within the package -->
-                    <from class="teammates.test.pageobjects.*" />
-                </allow>
-                <allow>
-                    <!-- allow dependency from UI tests -->
-                    <from class="teammates.test.cases.browsertests.*" />
-                </allow>
-            </deny>
-        </access-rule>
+    
+    <ruleset name="Architectural rules for test code">
+        <ruleset name="TestDriverComponent: Test cases should not be dependent on each other">
+            <access-rule>
+                <deny>
+                    <from class="${testCases}.**" />
+                    <to class="${testCases}.**" />
+                    <allow>
+                        <!-- allow dependency from inner class -->
+                        <from class="${testCases}.**$**" />
+                    </allow>
+                    <allow>
+                        <to>
+                            <!-- allow dependency to inner class -->
+                            <include class="${testCases}.**$**" />
+                            <!-- allow dependency to base class -->
+                            <include class="${testCases}.**Base**" />
+                        </to>
+                    </allow>
+                    <allow>
+                        <!-- allow dependency from Feedback*QuestionUiTest to FeedbackQuestionUiTest -->
+                        <from class="${testCases}.browsertests.Feedback*QuestionUiTest" />
+                        <to class="${testCases}.browsertests.FeedbackQuestionUiTest" />
+                    </allow>
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="TestDriverComponent: Only UI tests can access page object classes">
+            <access-rule>
+                <deny>
+                    <from class="${test}.**" >
+                        <exclude class="${test}.pageobjects.**" /> <!-- allow dependency within the package -->
+                    </from>
+                    <to class="${test}.pageobjects.**" />
+                    <allow>
+                        <from class="${testCases}.browsertests.**" />
+                    </allow>
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="TestDriverComponent: Only certain test cases can test storage:entity">
+            <access-rule>
+                <deny>
+                    <from class="${test}.**" />
+                    <to class="${storage}.entity.**" />
+                    <allow>
+                        <from>
+                            <include class="${testCases}.datatransfer.**" />
+                            <include class="${testCases}.storage.**" />
```

Comment:
I have checked the `FeedbackResponseCommentDbTest` and I agree with you that it should not have the dependency at all. Go ahead and make the necessary changes, and remove this exclusion afterwards.